### PR TITLE
Add assigner for crawl tasks [ZZ-45]

### PR DESCRIPTION
**Pull request by batanoffs**
Added constant `CRAWL_TASK_STATUS` to store the status for the tasks, please give any suggestions if it should be placed else where . Check line 13 to 20;

### CODE REVIEW

File: src/assigner.ts [hunk 1]
```diff
@@ -1,14 +1,94 @@
...
+import logger from "./utils/logger";
+
+// Declare and define constant for crawl task status.
+// Better to move it but where to? Suggestion db/prisma/constants - file constants.ts
+const CRAWL_TASK_STATUS = {
+    PENDING: "PENDING",
+    IN_PROGRESS: "IN_PROGRESS",
+    COMPLETED: "COMPLETED",
+    FAILED: "FAILED",
+};
```
Comment (DanielDimitrov5) [inline src/assigner.ts]: Import this 'import { CrawlTaskStatus } from "@prisma/client"'
This is the enum type from the schema. From there you can access the statuses directly.
No need for this.

File: src/assigner.ts [hunk 2]
```diff
@@ -1,14 +1,94 @@
...
+              1000
+            : Infinity;
+
+        // Check if it's time to scrape again
+        if (timeSinceLastScrape >= source.scrape_frequency_seconds) {
+
+            // Check if crawl task is created for current source
+            const getTask = await prisma.crawlTasks.findFirst({
+                where: {
+                    source: {
```
Comment (DanielDimitrov5) [inline src/assigner.ts]: This table is called 'Sources'.
Did you even test it?

File: src/assigner.ts [hunk 3]
```diff
@@ -1,14 +1,94 @@
...
+                },
+            });
+
+            // If task is found
+            if (getTask) {
+
+                // Check if status is NOT pending or in-progress and create new one
+                if (
+                    getTask.status !== CRAWL_TASK_STATUS.PENDING ||
+                    getTask.status !== CRAWL_TASK_STATUS.IN_PROGRESS
```
Comment (DanielDimitrov5) [inline src/assigner.ts]: Check this logic once again.

**Review by DanielDimitrov5 — CHANGES_REQUESTED** [review]
Test the assigner when the issues are fixed.
When you start, it should create just 3 tasks and nothing more.

**Comment by batanoffs** [comment]
Changes:
- Update logic for assigner as requested. 
- Added try/catch. 
- Updated logger to work with numbers instead of bool

During tests:
- my code didn't generate new warnings
- tested and created 3 tasks, when when status is changed to "COMPLETE", it generated a new one

For review - at line 37:
```js 
const getTask = await prisma.crawlTasks.findFirst({
       where: {
                 status: CrawlTaskStatus.PENDING || CrawlTaskStatus.IN_PROGRESS,
                 source_id: source.id,
               },
       });
```
Should i use findMany instead of findFirst?

**Review by DanielDimitrov5 — APPROVED** [review]
Perfect one!

**Pull request merged**